userspace/libsinsp/container.cpp

```diff
@@ -517,6 +517,26 @@ void sinsp_container_manager::create_engines()
 #endif
 }
 
+void sinsp_container_manager::update_container_with_size(sinsp_container_type type,
+							 const std::string& container_id)
+{
+	for(auto& eng : m_container_engines)
```

Comment:
We really want a pointer in the container back to the engine it seems

Reply:
yeah I was explicitly trying to avoid that... I want the data to stay data; mostly so we can mock the producer

I also don't really want everyone that uses container_info to think it's ok to go straight to the engine instead of going through the container_manager. Maybe that isn't the right opinion.

Reply:
I changed this to have a seperate map that tracks the engines by type. Tell me what you think.